posthog-react-native/src/types.ts

```diff
@@ -51,3 +51,8 @@ export interface PostHogCustomAsyncStorage {
   getItem: (key: string) => Promise<string | null>
   setItem: (key: string, value: string) => Promise<void>
 }
+
+export interface PostHogCustomSyncStorage {
+  getItem: (key: string) => string | null
+  setItem: (key: string, value: string) => void
+}
```

Comment:
Not redundant?

I mean, maybe the type of the getItem on the above `PostHogCustomAsyncStorage` is wrong, and maybe should be a single one with an added optional method  `preloadPromise`.

Then both sync and async storages classes can just implement this storage, instead of extending or whatever